### DISCUSSION TITLE
Stake pool metadata hash length in test vectors

### DISCUSSION THREAD

**Discussion by mkazlauskas** [CardanoSolutions/ogmios#278] [Q & A] [answered]
Pool registration metadata hash [has a length of 20 in the test vectors](https://github.com/CardanoSolutions/ogmios/search?q=62797465737472696e67), while [CDDL says it's type is $hash32](https://github.com/input-output-hk/cardano-ledger/blob/master/eras/babbage/test-suite/cddl-files/babbage.cddl#L403) (should have length of 64). Is Ogmios using a different (non base16) encoding for it?

**Comment by KtorZ** [comment, marked as answer]
Indeed. Ogmios' tests rely on arbitrary generator coming from the consensus and ledger code bases. Sadly however, some of those generators aren't completely _faithful_ when it comes to hashes (stake pool metadata hash being one of them). While this could be patched on-the-fly in Ogmios, I think that the proper way to fix it is rather upstream, in cardano-ledger.
